apel/db/records/record.py

```diff
@@ -380,3 +430,22 @@ def _check_fields(self):
                 elif value is not None:
                     raise InvalidRecordException("Datetime field " + key +
                                     " doesn't contain an datetime.")
+
+    def _extract_benchmark_dict(self, fielddict, field):
+        """Extract a preferrred benchmark type and value from a fielddict"""
+
+        benchmark_priority = ("hepscore23", "hepspec", "si2k")
+
+        if field in fielddict:
+            try:
+                cleaned_dict = self._clean_up_dict(fielddict[field])
+                # Covert keys to lower case
+                cleaned_dict = {k.lower(): v for k, v in cleaned_dict.items()}
+            except ValueError as e:
+                raise InvalidRecordException("Expecting dictionary-like value. %s" % e)
+
+            for benchmark_type in benchmark_priority:
+                if benchmark_type in cleaned_dict:
+                    return benchmark_type, cleaned_dict[benchmark_type]
+            else:
+                raise InvalidRecordException("No valid benchmark type found")
```

Comment:
I think these lines should be indented so that the else is under the purview of the if, not the for.  It's dead code right now.

Reply:
Ah, yes, I think you're right.